discovery-infra/test_infra/logs_utils.py

```diff
@@ -24,17 +25,18 @@ def verify_logs_uploaded(cluster_tar_path, expected_min_log_num, installation_su
                     _verify_bootstrap_logs_uploaded(tempdir, gz, installation_success)
                 elif "master" in gz or "worker" in gz:
                     _verify_node_logs_uploaded(tempdir, gz)
+                elif "controller" in gz:
+                    if check_oc:
+                        _verify_oc_logs_uploaded(os.path.join(tempdir, gz))
```

Comment:
Why not to have a function `_verify_oc_logs_uploaded` that get a parameter `installation_success` and check logs on success and on failure?

We are currently missing checking controller logs on success, I would suggest something like:
```
elif "controller" in gz:
    _verify_controller_logs_uploaded(cluster_tar_path, installation_success)
 
 def _verify_controller_logs_uploaded(cluster_tar_path, installation_success):
      # check that the controller logs exist (controller.logs)
      if !installation_success:
          # check that all the subdirectories of must-gather exists (the same way it is done for installer-gather)
```

Reply:
we do check controller logs on success through the check of min_num_of_logs. In addition there is no point to check within the must-gather dirctory since they are too many directories there and with temporary names. if the outer zip exists it means that must-gather tool ran successfully. Otherwise the zip is not created (unlike other log cases)